### PR TITLE
chore: replace use of `AddressBook` with `Roster` in `ReconnectMetrics and `AddressBookMetrics`

### LINKED ISSUE #16718: Replace use of `AddressBook` with `Roster` in Metrics
**Background**
Several metrics are initialized with `AddressBook`.  Replace this with the usage of `Roster` in the following metrics: 
1. `ReconnectMetrics` in `SyncGossip.java`
2. `AddressBookMetrics` in `SwirldsPlatform.java`

**Acceptance Criteria**

The metrics still work.   

**Dependencies**
None

### PR DESCRIPTION
**Pull request by edward-swirldslabs**
**Description**:

A small PR to replace the use of `AddressBook` with `Roster` in `ReconnectMetrics` and `AddressBookMetrics`. 

**Related issue(s)**:

Fixes #16718

### CODE REVIEW

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java [hunk 1]
```diff
@@ -222,7 +222,7 @@ public SwirldsPlatform(@NonNull final PlatformComponentBuilder builder) {
 
         platformWiring = new PlatformWiring(platformContext, blocks.model(), blocks.applicationCallbacks());
 
-        registerAddressBookMetrics(platformContext.getMetrics(), currentAddressBook, selfId);
+        registerAddressBookMetrics(platformContext.getMetrics(), currentRoster, selfId);
```
Comment (anthony-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:225]: Minor: might be worth renaming the method too.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ReconnectMetrics.java [hunk 2]
```diff
@@ -147,7 +146,7 @@ public void incrementReceiverEndTimes() {
      *
      * @param nodeId the peer being rejected.
      */
-    public void recordReconnectRejection(@NonNull final NodeId nodeId) {
+    public void recordReconnectRejection(@NonNull final Long nodeId) {
```
Comment (anthony-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/metrics/ReconnectMetrics.java:149]: Minor: might use a primitive `long`.

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookMetrics.java [hunk 3]
```diff
@@ -33,19 +33,19 @@ private AddressBookMetrics() {}
...
                 .withDescription("The node ID number of this member"));
 
-        metrics.getOrCreate(
-                new FunctionGauge.Config<>(Metrics.INFO_CATEGORY, "members", Integer.class, addressBook::getSize)
-                        .withUnit("count")
-                        .withDescription("total number of nodes currently in the address book"));
+        metrics.getOrCreate(new FunctionGauge.Config<>(
+                        Metrics.INFO_CATEGORY, "members", Integer.class, roster.rosterEntries()::size)
+                .withUnit("count")
+                .withDescription("total number of nodes currently in the address book"));
```
Comment (anthony-swirldslabs) [inline platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/address/AddressBookMetrics.java:49]: Minor: s/address book/roster/

Also minor, might want to rename this class to `RosterMetrics` as well. That's what I originally did at https://github.com/hashgraph/hedera-services/pull/16026

**Review by anthony-swirldslabs — DISMISSED** [review]

File: platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java (hunk 1 above)
Comment (edward-swirldslabs) [reply platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/SwirldsPlatform.java:225]: renamed the method

**Review by anthony-swirldslabs — APPROVED** [review]

**Pull request merged**